app/models/state_file_az_intake.rb

```diff
@@ -155,7 +159,16 @@ def disqualifying_eligibility_rules
   def ask_whether_incarcerated?
     has_valid_ssn = primary.ssn.present? && !primary.has_itin?
     has_valid_agi = direct_file_data.fed_agi <= (filing_status_mfj? || filing_status_hoh? ? 25_000 : 12_500)
-    has_valid_ssn && has_valid_agi
+    if filing_status_mfj?
+      spouse_has_valid_ssn = spouse.ssn.present? && !spouse.has_itin?
+      has_valid_ssn && has_valid_agi && spouse_has_valid_ssn
+    else
+      has_valid_ssn && has_valid_agi
+    end
+  end
+
+  def qualified_for_excise_credit?
+    was_incarcerated_no? && ssn_no_employment_no? && household_excise_credit_claimed_no?
```

Comment:
These two also may be good to add unit tests for.

Reply:
these all have unit tests in az140 spec